exporters/otlp/otlpmetric/exporter.go

```diff
@@ -0,0 +1,104 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build go1.18
+// +build go1.18
+
+package otlpmetric // import "go.opentelemetry.io/otel/exporters/otlp/otlpmetric"
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/internal/transform"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+	mpb "go.opentelemetry.io/proto/otlp/metrics/v1"
+)
+
+// exporter exports metrics data as OTLP.
+type exporter struct {
+	// Ensure synchronous access to the client across all functionality.
+	clientMu sync.Mutex
+	client   Client
+
+	shutdownOnce sync.Once
+}
+
+// Export transforms and transmits metric data to an OTLP receiver.
+func (e *exporter) Export(ctx context.Context, rm metricdata.ResourceMetrics) error {
+	otlpRm, tErr := transform.ResourceMetrics(rm)
+	// Best effort upload of transformable metrics.
+	e.clientMu.Lock()
+	upErr := e.client.UploadMetrics(ctx, otlpRm)
```

Comment:
Should we ever skip uploading if there are no metrics to send?

Reply:
This was a question I had as well. Not sending an empty group of metrics will prevent potentially unneeded network overhead, but it might also not communicate a lack of measurements which might be meaningful to a receiver. I saw this "lack of measurement is meaningful" issue as a concern for a receiver or the SDK itself and I wanted to make the exporter as "dumb" as possible regarding this.

I'm happy to change the behavior if you think this reasoning is misplaced.

Reply:
That makes sense.  I agree this should be a concern of the SDK itself.  If the MetricReader isn't filtering out empty batches, we probably shouldn't do that here either.